pkg/rancher-desktop/pages/containers/console/_id.vue

```diff
@@ -0,0 +1,781 @@
+import { WebLinksAddon } from '@xterm/addon-web-links';
+import { SearchAddon } from '@xterm/addon-search';
+
+import LoadingIndicator from '@pkg/components/LoadingIndicator.vue';
+import { ContainerEngine } from '@pkg/config/settings';
+import { ipcRenderer } from '@pkg/utils/ipcRenderer';
+
+let logInterval = null;
+
+export default Vue.extend({
+  name: 'ContainerConsole',
+  title: 'Container Console',
+  components: {
+    BadgeState,
+    Banner,
+    LoadingIndicator,
+  },
+  data() {
+    return {
+      settings: undefined,
+      ddClient: null,
+      isLoading: true,
+      error: null,
+      containerName: '',
+      containerState: '',
+      isContainerRunning: false,
+      terminal: null,
+      fitAddon: null,
+      searchAddon: null,
+      lastLogTimestamp: null,
+      pendingLogs: '',
+      isSearchExpanded: false,
+      searchTerm: '',
+    };
+  },
+  computed: {
+    ...mapGetters('k8sManager', { isK8sReady: 'isReady' }),
+    containerId() {
+      return this.$route.params.id;
+    },
+    isNerdCtl() {
+      return this.settings?.containerEngine?.name === ContainerEngine.CONTAINERD;
+    },
+    selectedNamespace() {
+      return this.settings?.containers?.namespace;
+    },
+  },
+  async mounted() {
+    this.$store.dispatch('page/setHeader', {
+      title: this.t('containers.console.title'),
+      description: '',
+    });
+
+    ipcRenderer.on('settings-read', (event, settings) => {
+      this.settings = settings;
+      this.initializeConsole();
+    });
+
+    ipcRenderer.send('settings-read');
+    this.initializeConsole();
+
+    // Add global keyboard shortcut for search
+    window.addEventListener('keydown', this.handleGlobalKeydown);
+  },
+  beforeDestroy() {
+    this.stopStreaming();
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+    ipcRenderer.removeAllListeners('settings-read');
+    window.removeEventListener('keydown', this.handleGlobalKeydown);
+  },
+  methods: {
+    async initializeConsole() {
+      if (window.ddClient && this.isK8sReady && this.settings) {
+        this.ddClient = window.ddClient;
+        await this.getContainerInfo();
+        await this.fetchLogs();
+
+        if (this.isContainerRunning) {
+          this.startStreaming();
+        }
+      }
+    },
+    async getContainerInfo() {
+      try {
+        const listOptions = { all: true };
+
+        if (this.isNerdCtl && this.selectedNamespace) {
+          listOptions.namespace = this.selectedNamespace;
+        }
+
+        const containers = await this.ddClient?.docker.listContainers(listOptions);
+
+        const container = containers.find(c => c.Id === this.containerId || c.Id.startsWith(this.containerId));
+
+        if (container) {
+          const names = Array.isArray(container.Names) ? container.Names : container.Names.split(/\s+/);
+          this.containerName = names[0]?.replace(/_[a-z0-9-]{36}_[0-9]+/, '') || container.Id.substring(0, 12);
```

Comment:
Given that we don't care about `names` other than the first, wouldn't it make more sense to just grab that?
```suggestion
          const name = Array.isArray(container.Names) ? container.Names[0] : container.Names.split(/\s+/)?.[0];
          this.containerName = name?.replace(/_[a-z0-9-]{36}_[0-9]+/, '') || container.Id.substring(0, 12);
```